src/WasmSdk/Sdk/Sdk.targets

```diff
@@ -15,6 +15,18 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Import Sdk="Microsoft.NET.Sdk.StaticWebAssets" Project="Sdk.targets" Condition="'$(RuntimeIdentifier)' != 'wasi-wasm' and '$(_WasmSdkImportsMicrosoftNETSdkStaticWebAssets)' == 'true'" />
   <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.targets" Condition="'$(RuntimeIdentifier)' == 'wasi-wasm' and '$(_WasmSdkImportsMicrosoftNetSdk)' == 'true'" />
 
+  <PropertyGroup>
+    <!-- Implicit HotReload package reference for .NET 10+ if not explicitly turned off -->
+    <_WasmEnableHotReload>$(WasmEnableHotReload)</_WasmEnableHotReload>
+    <_WasmEnableHotReload Condition="'$(_WasmEnableHotReload)' == '' and '$(Configuration)' == 'Release'">false</_WasmEnableHotReload>
+    <_WasmEnableHotReload Condition="'$(_WasmEnableHotReload)' == '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals('$(TargetFrameworkVersion)', '10.0'))">true</_WasmEnableHotReload>
+  </PropertyGroup>
+  <Target Name="_ImplicitlyReferenceHotReload" BeforeTargets="ProcessFrameworkReferences">
+    <ItemGroup Condition="'$(_WasmEnableHotReload)' == 'true'">
+      <PackageReference Include="Microsoft.DotNet.HotReload.WebAssembly.Browser" Version="$(NETCoreSdkVersion)" />
+    </ItemGroup>
+  </Target>
```

Comment:
Will this get trimmed away?

An alternative would be not having a package at all and instead inject the dll and the initializer in a target as part of the build. AFAIK, it doesn't even need to be part of the compilation, does it? 

As long as we tell the wasm runtime to load the dll when we are bootstrapping the framework we should be good, shouldn't we?